provider/pkg/provider/agent_pool.go

```diff
@@ -31,6 +32,9 @@ func GenerateAgentPoolProperties(input PulumiServiceAgentPoolInput, agentPool pu
 	if input.Description != "" {
 		inputMap["description"] = resource.NewPropertyValue(input.Description)
 	}
+	if input.ForceDestroy {
+		inputMap["forceDestroy"] = resource.NewPropertyValue(input.ForceDestroy)
```

Comment:
Might want to make `forceDestroy` a constant

Reply:
Hmm, not sure it makes sense to make one input a constant and not any of the others... so I'd leave that to a future optimization since this is not the pattern anywhere in the provider.